mail/views.py

```diff
@@ -104,20 +132,17 @@ def post(self, request, *args, **kargs):  # pylint: disable=unused-argument
                     email_body=email_body,
                     sender_name=sender_name,
                 )
-
         mailgun_responses = MailgunClient.send_batch(
             subject=email_subject,
             body=email_body,
             recipients=emails,
             sender_name=sender_name,
         )
         return Response(
-            status=status.HTTP_200_OK,
+            status=_make_batch_status(mailgun_responses),
             data={
-                "batch_{}".format(batch_num): {
-                    "status_code": resp.status_code,
-                    "data": generate_mailgun_response_json(resp)
-                } for batch_num, resp in enumerate(mailgun_responses)
+                "batch_{}".format(batch_num): _make_batch_response_dict(resp, exception)
+                for batch_num, (_, resp, exception) in enumerate(mailgun_responses)
```

Comment:
Why is `data` a dictionary, rather than a list? Especially since the keys of the dictionary are just strings for the form "batch_{num}" -- seems to me like it really wants to be a list. Why not simply return `data=mailgun_responses`, and not process it at all?